src/main/java/org/folio/circulation/support/fetching/PageableFetcher.java

```diff
@@ -0,0 +1,91 @@
+package org.folio.circulation.support.fetching;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.circulation.support.http.client.Offset.noOffset;
+import static org.folio.circulation.support.http.client.PageLimit.limit;
+import static org.folio.circulation.support.results.Result.failed;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.domain.MultipleRecords;
+import org.folio.circulation.support.ServerErrorFailure;
+import org.folio.circulation.support.http.client.CqlQuery;
+import org.folio.circulation.support.http.client.Offset;
+import org.folio.circulation.support.http.client.PageLimit;
+import org.folio.circulation.support.results.Result;
+import org.slf4j.Logger;
+
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor
+public final class PageableFetcher<T> {
```

Comment:
I made an aborted attempt at writing something like this a while ago. Thanks for picking up the mantle to write this.